crucible-mir/src/Mir/Trans.hs

```diff
@@ -1364,8 +1364,23 @@ evalRval rv@(M.RAdtAg (M.AdtAg adt agv ops ty optField)) = do
             case adt^.adtkind of
                 M.Struct -> buildStruct adt es
                 M.Enum _ -> buildEnum adt (fromInteger agv) es
-                M.Union -> do
-                    mirFail $ "evalRval: Union types are unsupported, for " ++ show (adt ^. adtname)
+                -- This is a hack. crucible-mir doesn't _actually_ support union
+                -- types, and if you try simulating a code that performs an
+                -- operation on a union-typed value, then it will fail at
+                -- simulation time. Nevertheless, we intentionally choose to
+                -- give unions a dummy value here instead of outright failing so
+                -- that crucible-mir can translate some union-typed declarations
+                -- without crashing during MIR translation.
+                --
+                -- We apply `packAny` here to match the similar hack described
+                -- in the union case of `Mir.Trans.tyToRepr`, which gives
+                -- union-typed values a dummy representation of `AnyRepr`.
+                --
+                -- See #1429 for the larger issue of properly supporting union
+                -- types.
+                M.Union ->
+                  let dummyExpr = R.App $ usizeLit 0xdeadbeef
+                   in pure $ packAny $ MirExp knownRepr dummyExpr
```

Comment:
I mentioned in https://github.com/GaloisInc/crucible/issues/1534#issuecomment-3285856653 that we might want to make this dummy expression the same size as the union it replaces, since we have access to that size here, but I realized I wasn't sure when/why that would be necessary, so I left it as a simple `usizeLit`. I welcome opinions on whether this is appropriate.

Reply:
> I think this will matter if you put a union value in a data structure that is represented as a `MirAggregate`

All of the `MirAggregate` operations take separate size, offset, and `TypeRepr` arguments.  They don't try to compute the size from the `TypeRepr`, but instead compute all three from `Mir.Ty` info.